src/pages/Scholarships.jsx

```diff
@@ -33,54 +34,48 @@ function ScholarshipsPage() {
   const location = useLocation();
   const history = useHistory();
 
-  // sets min and max amount for the amount filter
-  const setMinMax = (index, val) => {
+  const params = queryString.parse(location.search, { parseNumbers: true });
+  const { minAmount, maxAmount } = params;
+
+  const setQueryParam = (index, val) => {
     history.push({
       search: queryString.stringify({
-        ...queryString.parse(location.search),
+        ...params,
         [index]: val,
       }),
     });
   };
 
-  const params = queryString.parse(location.search, { parseNumbers: true });
-  const minParam = params.minAmount;
-  const maxParam = params.maxAmount;
-
-  if ((minParam && !Number.isInteger(minParam)) || minParam === 0) {
-    delete params.minAmount;
-    history.replace({ search: queryString.stringify(params) });
-  }
-
-  if ((maxParam && !Number.isInteger(maxParam)) || maxParam === 0) {
-    delete params.maxAmount;
-    history.replace({ search: queryString.stringify(params) });
-  }
+  const clearQueryParam = (index) => {
+    const newParams = { ...params };
+    delete newParams[index];
+    history.replace({ search: queryString.stringify(newParams) });
+  };
 
   useEffect(() => {
     // TODO: Create cancellable promises
     Scholarships.list({ sortField, sortDir })
       .then((results) => {
-        const filterError = maxParam > 0 && maxParam <= minParam;
-
-        if (filterError) setScholarships(results);
-        else
+        if (maxAmount > 0 && maxAmount <= minAmount)
+          setError('The minimum amount must be less than the Maximum.');
```

Comment:
You can do this check before `Scholarships.list()` is called. Also you can chain a `.then()` to this `.then()` that calls `setError(null)` so that this `.then()` can be a single statement.